app/models/configuration_script.rb

```diff
@@ -1,3 +1,20 @@
 class ConfigurationScript < ActiveRecord::Base
-  belongs_to :configuration_manager, :class_name => 'ManageIQ::Providers::ConfigurationManager'
+  belongs_to  :manager,
+              :class_name  => 'ManageIQ::Providers::ConfigurationManager',
+              :foreign_key => :configuration_manager_id
+  include ProviderObjectMixin
+
+  def run
+    with_provider_object do |jt|
+      jt.launch('extra_vars' => variables)
```

Comment:
variables are the extra_vars pulled in from the JobTemplates on the tower machine. I can make a change to also allow them to be passed in as an argument.